nixos/modules/services/security/crowdsec.nix

```diff
@@ -0,0 +1,959 @@
+            UMask = "0077";
+            ProtectKernelTunables = true;
+            ProtectKernelModules = true;
+            ProtectControlGroups = true;
+            ProtectProc = "invisible";
+            SystemCallFilter = [
+              " " # This is needed to clear the SystemCallFilter existing definitions
+              "~@reboot"
+              "~@swap"
+              "~@obsolete"
+              "~@mount"
+              "~@module"
+              "~@debug"
+              "~@cpu-emulation"
+              "~@clock"
+              "~@raw-io"
+              "~@privileged"
+              "~@resources"
+            ];
+            CapabilityBoundingSet = [
+              " " # Reset all capabilities to an empty set
+              "CAP_SYSLOG" # Add capability to read syslog
+            ];
+            RestrictAddressFamilies = [
+              " " # This is needed to clear the RestrictAddressFamilies existing definitions
+              "none" # Remove all addresses families
+              "AF_UNIX"
+              "AF_INET"
+              "AF_INET6"
+            ];
+            DevicePolicy = "closed";
+            ProtectKernelLogs = true;
+            SystemCallArchitectures = "native";
+            DynamicUser = true;
+            RestrictNamespaces = true;
+            RestrictRealtime = true;
+            RestrictSUIDSGID = true;
+            ExecReload = [
+              " " # This is needed to clear the ExecReload definitions from upstream
+            ];
+            ExecStart = [
+              " " # This is needed to clear the ExecStart definitions from upstream
+              "${lib.getExe' cfg.package "crowdsec"} -c ${configFile} -info"
+            ];
+            ExecStartPre = [
+              " " # This is needed to clear the ExecStartPre definitions from upstream
+              "${lib.getExe setupScript}"
+              "${lib.getExe' cfg.package "crowdsec"} -c ${configFile} -t -error"
+            ];
+          };
+        };
+      };
+
+      systemd.tmpfiles.settings = {
+        "10-crowdsec" =
+
+          builtins.listToAttrs (
+            map
+              (dirName: {
+                inherit cfg;
+                name = lib.strings.normalizePath dirName;
+                value = {
+                  d = {
+                    user = cfg.user;
+                    group = cfg.group;
+                    mode = "0750";
+                  };
+                };
+              })
+              [
+                stateDir
+                hubDir
+                confDir
+                localScenariosDir
+                localPostOverflowsDir
+                localPostOverflowsS01WhitelistDir
+                parsersDir
+                localParsersS00RawDir
+                localParsersS01ParseDir
+                localParsersS02EnrichDir
+                localContextsDir
+                notificationsDir
+                pluginDir
+              ]
+          )
+          // builtins.listToAttrs (
```

Comment:
I _suspect_ it may be simpler (and more robust) to use `mkMerge` in this `systemd.tmpfiles.settings` definition, instead of using `//` updates.

One case `mkMerge` handles better is when the same attrname is defined by both sides. `//` would simply shadow the LHS with the RHS's value, while `mkMerge` would merge the two definitions (or throw a sane error if that's not possible).

Reply:
I see your point but I think that I'm not experienced enough to give a strong opinion about this... but I'd say that using `mkMerge` may be the more intuitive one.
However, I'd wait for another opinion who is more experienced than me :sweat_smile: